pkg/controller/backup_controller.go

```diff
@@ -230,6 +234,22 @@ func (b *backupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	switch original.Status.Phase {
 	case "", velerov1api.BackupPhaseNew:
 		// only process new backups
+	case velerov1api.BackupPhaseInProgress:
+		// if backup is in progress, we should not process it again
+		// we want to mark it as failed to avoid it being stuck in progress
+		// if this backupReconciler uuid is the same as the one in the backup, then we are the one who is processing it
+		// if so, mark it as failed, last loop did not successfully complete the backup
+		if original.Annotations[velerov1api.BackupControllerUID] == string(b.uuid) {
+			log.Debug("Backup was processed by this backupReconciler and still has in progress status upon reconcile, marking it as failed")
+		}
+		failedCopy := original.DeepCopy()
+		failedCopy.Status.Phase = velerov1api.BackupPhaseFailed
+		failedCopy.Status.FailureReason = "Backup from previous reconcile still in progress"
+		if err := kubeutil.PatchResource(original, failedCopy, b.kbClient); err != nil {
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, errors.Wrapf(err, "error updating Backup status to %s", failedCopy.Status.Phase)
+		}
+		// patch to mark it as failed succeeded, do not requeue
+		return ctrl.Result{}, nil
```

Comment:
Also, should there be an else clause where we *do* requeue when the reconciler UID doesn't match?

Reply:
++

Reply:
When reconciler UID doesn't match, the backup may belong to a reconciler that is still processing the backup.

Reply:
So we don't requeue if UID doesn't match.. unless a way to check if another reconciler UID exists.

Reply:
Hmm. But if we don't requeue, then doesn't that mean the backup is still stuck. It's InProgress, and nothing else will modify it.

Reply:
But yes -- there are 2 situations possible here when UID doesn't match:
1) status update actually failed earlier, nothing is processing it.
2) backup still running in other reconciler, got queued because of an update to the backup.

For 1), if we don't requeue, we're stuck forever, if we do requeue, we eventually hit the right reconciler.
For 2), if we don't requeue, backup completes normally, but if we do, then we'll keep having to requeue on no match, on and on (waste of resources, but otherwise no harm done).

Reply:
We're solving for case where velero pod has not restarted. we only want the one thread who was processing it that failed patching to come back and retry patching to fail later.

Reply:
It still seems that if we don't requeue here, then that one thread will never retry. We already requeued, a different thread picked it up, so we bailed, but if we don't requeue now, then I suspect that the backup will still be stuck InProgress forever, just as without this patch.  The more threads we have, the smaller the likelihood that the first requeue will come to the one we want, so if we don't requeue here, especially if there are more than 2 or 3 threads, not very many of these failed-status-update backups will actually go into the failed state.